packages/tools/src/bn.ts

```diff
@@ -182,67 +163,70 @@ export class BN {
     return this.#bn.isZero()
   }
 
+  public get value(): string {
+    return this.#bn.toString()
+  }
+
+  public clone(cfg?: BnCfg): BN {
+    return new BN(this.value, { ...this.#cfg, ...(cfg || {}) })
+  }
+
   public mul(other: BN): BN {
-    const numWithGreatestDecimals = BN.#getGreatestDecimal(this, other)
+    const greatestDecimals = BN.#getGreatestDecimal(this, other).decimals
     const [numA, numB] = BN.#toGreatestDecimals(this, other)
 
     return new BN(
       numA.bn
         .multipliedBy(numB.bn)
-        .dividedBy(BN.#instance(10).pow(numWithGreatestDecimals.cfg.decimals)),
+        .dividedBy(BN.#instance(TEN_POWER).pow(greatestDecimals)),
       {
         ...this.#cfg,
-        decimals: numWithGreatestDecimals.cfg.decimals,
+        decimals: greatestDecimals,
       },
     )
   }
 
   public div(other: BN): BN {
-    if (other.bn.isZero())
-      throw new TypeError(`Cannot divide ${other.valueOf()} by zero`)
+    if (other.isZero) {
+      throw new TypeError(`Cannot divide ${other.value} by zero`)
+    }
 
-    const numWithGreatestDecimals = BN.#getGreatestDecimal(this, other)
+    const greatestDecimals = BN.#getGreatestDecimal(this, other).decimals
     const [numA, numB] = BN.#toGreatestDecimals(this, other)
 
-    return new BN(numA.bn.dividedBy(numB.bn), {
+    const fr = BN.#instance(TEN_POWER).pow(greatestDecimals)
+
+    return new BN(numA.bn.dividedBy(numB.bn).multipliedBy(fr), {
       ...this.#cfg,
-      decimals: numWithGreatestDecimals.cfg.decimals,
+      decimals: greatestDecimals,
     })
   }
 
   public add(other: BN): BN {
-    const numWithGreatestDecimals = BN.#getGreatestDecimal(this, other)
     const [numA, numB] = BN.#toGreatestDecimals(this, other)
 
     return new BN(numA.bn.plus(numB.bn), {
       ...this.#cfg,
-      decimals: numWithGreatestDecimals.cfg.decimals,
+      decimals: BN.#getGreatestDecimal(this, other).decimals,
     })
   }
 
   public sub(other: BN): BN {
-    const numWithGreatestDecimals = BN.#getGreatestDecimal(this, other)
     const [numA, numB] = BN.#toGreatestDecimals(this, other)
 
     return new BN(numA.bn.minus(numB.bn), {
       ...this.#cfg,
-      decimals: numWithGreatestDecimals.cfg.decimals,
+      decimals: BN.#getGreatestDecimal(this, other).decimals,
     })
   }
 
   public pow(other: number): BN {
-    return new BN(
-      this.#bn
-        .pow(BN.#instance(other))
-        .dividedBy(
-          BN.#instance(10).pow(
-            BN.#instance(this.#cfg.decimals).multipliedBy(
-              BN.#instance(other - 1),
-            ),
-          ),
-        ),
-      this.#cfg,
+    const bn = BN.#instance
+    const fr = bn(TEN_POWER).pow(
+      bn(this.#cfg.decimals).multipliedBy(bn(other - 1)),
```

Comment:
maybe it'll be more convenient to make method, returning tenth/one-tenth in power of argument?
`getTenthPowerIn(arg)` or smth